### PR TITLE
Fix copyright in MSFT VersionInfo

### PR DESCRIPTION
**Pull request by richsalz**
Also add rc file to openssl app

Thanks to user RTT for pointing this out.

### CODE REVIEW

File: apps/build.info [hunk 1]
```diff
@@ -23,6 +23,11 @@ IF[{- !$disabled{apps} -}]
   INCLUDE[openssl]=.. ../include
   DEPEND[openssl]=libapps.a ../libssl
 
+IF[{- $config{target} =~ /^(?:Cygwin|mingw|VC-)/ -}]
+  GENERATE[openssl.rc]=util/mkrc.pl openssl
+  SOURCE[openssl]=openssl.rc
+ENDIF
```
Comment (levitte) [inline apps/build.info]: Appveyor shows us that this creates a "surprising" disturbance...  [this section in apps/progs.pl](https://github.com/openssl/openssl/blob/master/apps/progs.pl#L24-L26) needs to be rewritten by inserting a grep in the middle, like this:

``` perl
diff --git a/apps/progs.pl b/apps/progs.pl
index f832467b57..f16bc10fac 100644
--- a/apps/progs.pl
+++ b/apps/progs.pl
@@ -23,6 +23,7 @@ my $apps_openssl = shift @ARGV;
 # the lookups in %unified_info
 my @openssl_source =
     map { @{$unified_info{sources}->{$_}} }
+    grep { /\.o$/ }
         @{$unified_info{sources}->{$apps_openssl}};
 
 foreach my $filename (@openssl_source) {
```

Comment (richsalz) [reply apps/build.info]: done, and also updated copyright year generation.  please reconfirm.

**Comment by richsalz** [comment]
I have a build recipe wrong.  Look at https://ci.appveyor.com/project/openssl/openssl/build/master.16660/job/o6at0feo36hmiesn#L1669 or https://ci.appveyor.com/project/openssl/openssl/build/master.16660/job/o6at0feo36hmiesn#L375

No equivalent line shows up for openssl.

Where and how does the output redirection happen?

What's wrong with apps/build.info?

**Comment by levitte** [comment]
Try this patch:

``` diff
diff --git a/Configure b/Configure
index 0934088d2d..ce33f488d6 100755
--- a/Configure
+++ b/Configure
@@ -2015,6 +2015,13 @@ EOF
                     $o = cleanfile($buildd, $o, $blddir);
                     $unified_info{sources}->{$ddest}->{$o} = 1;
                     $unified_info{sources}->{$o}->{$s} = 1;
+                } elsif ($s =~ /\.rc$/) {
+                    # We also recognise resource files
+                    my $o = $_;
+                    $o =~ s/\.rc$/.res/; # Resource configuration
+                    my $o = cleanfile($buildd, $o, $blddir);
+                    $unified_info{sources}->{$ddest}->{$o} = 1;
+                    $unified_info{sources}->{$o}->{$s} = 1;
                 } else {
                     $unified_info{sources}->{$ddest}->{$s} = 1;
                 }
```

**Comment by levitte** [comment]
What was wrong was that .rc files was only recognised in SHARED_SOURCE, not in SOURCE.  The diff I just submitted fixes that

**Comment by richsalz** [comment]
pushed and watching the CI's now ...

**Comment by richsalz** [comment]
it builds and passes (and was  confirmed by RTT), but I'll wait for an explicit reconfirm.

**Comment by richsalz** [comment]
Not backporting; fixed in master.

**Comment by bernd-edlinger** [comment]
Apparently this broke the mingw build:
```
Setting up wine1.6-amd64 (1:1.6.2-0ubuntu4) ...
Processing triggers for libc-bin (2.19-0ubuntu6.13) ...
Processing triggers for initramfs-tools (0.103ubuntu4.9) ...
update-initramfs: Not updating initramfs.
W: --force-yes is deprecated, use one of the options starting with --allow instead.
make depend && make _tests
make[1]: Entering directory `/home/travis/build/openssl/openssl'
make[1]: Leaving directory `/home/travis/build/openssl/openssl'
make[1]: Entering directory `/home/travis/build/openssl/openssl'
i686-w64-mingw32-gcc   apps/openssl.res.o   -o apps/openssl.res
/usr/lib/gcc/i686-w64-mingw32/4.8/../../../../i686-w64-mingw32/lib/../lib/libmingw32.a(lib32_libmingw32_a-crt0_c.o): In function `main':
/build/buildd/mingw-w64-3.1.0/build/i686-w64-mingw32-i686-w64-mingw32-crt/../../mingw-w64-crt/crt/crt0_c.c:18: undefined reference to `WinMain@16'
collect2: error: ld returned 1 exit status
make[1]: *** [apps/openssl.res] Error 1
make[1]: Leaving directory `/home/travis/build/openssl/openssl'
make: *** [tests] Error 2
+//// MAKE TEST FAILED
```

**Comment by bernd-edlinger** [comment]
Regarding generation of source files with copyright year from current time:
I hope this apps/progs.pl is not run automatically.

**Comment by richsalz** [comment]
Why do you hope it's not run automatically?
Looking at apps/build.info it looks like it's run whenever config is run (configdata.pm is new)

**Comment by richsalz** [comment]
I wonder why MingW broke with this, but the other rc files didn't cause an error.  Do you have a complete build log you can send me?

**Comment by bernd-edlinger** [comment]
> Why do you hope it's not run automatically?

It appears to me as if you have a book where every time you open
the book the copyright notice changes to the current year.

I am not a lawyer but to me that looks like it makes the whole copy-right
notice void.

**Comment by bernd-edlinger** [comment]
I don't have mingw myself, but I saw that in the travis build after this got merged:

i686-w64-mingw32-gcc:
https://travis-ci.org/openssl/openssl/jobs/356916268

i86_64-w64-mingw32-gcc:
https://travis-ci.org/openssl/openssl/jobs/356916269

**Comment by bernd-edlinger** [comment]
Yes, as I said I am not a lawyer...
It is just the first time I see an automatically adjusting copy-right.

**Comment by richsalz** [comment]
@dot-asm, @levitte any thoughts on the build-break Travis logs?
Why only apps/openssl.res ?  I could just remove the file?

**Comment by levitte** [comment]
Oh, interesting thing is that the resource file wasn't ever included when linking the .dlls on MinGW or Cygwin...  Looking at it, it looks like a very simple fix...

**Comment by richsalz** [comment]
Like just remove cygwin and mingw from the patterns?  I'll do a PR for that, if you're not already on it. :)

**Comment by levitte** [comment]
No, the other way around, make sure that the support in `Configurations/unix-Makefile.tmpl` is done right.

**Comment by levitte** [comment]
See #5730

**Pull request closed without merging**